commands/deploy.go

```diff
@@ -171,9 +171,24 @@ func (c *DeployCmd) Run(ctx *kong.Context, logWriters *LogWriters) (err error) {
 // IsValidNodeApp detects if a Node.js app is present in a given directory
 func IsValidNodeApp(dir string) (errs []error) {
 	packageJSONPath := filepath.Join(dir, "package.json")
-	if _, err := os.Stat(packageJSONPath); os.IsNotExist(err) {
-		errs = append(errs, fmt.Errorf("%s is not a file", packageJSONPath))
+	if _, err := os.Open(packageJSONPath); os.IsNotExist(err) {
+		log.Debug().Msg(fmt.Sprintf("[WARN] %s is not a file", packageJSONPath))
+	} else {
+		packageJSONContents, err := ioutil.ReadFile(packageJSONPath)
+		if err != nil {
+			log.Info().Err(err).Msg("Error reading your package.json")
+		}
```

Comment:
Calling `os.Open` and discarding the returned file handle is a resource leak. Just check if any error from `ioutil.ReadFile` represents a non-existent file. Try something like this
```suggestion
	if packageJSONContents, err := ioutil.ReadFile(packageJSONPath); err != nil {
		if os.IsNotExist(err) {
			log.Debug().Msg(fmt.Sprintf("[WARN] %s is not a file", packageJSONPath))
		} else {
			log.Info().Err(err).Msg("Error reading your package.json")
		}		
	} else {
```

Reply:
yeah i felt like there might have been a better solution to that. thanks